cmake/install-mod.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+# SPDX-Identifier: BSD-2-Clause
+
+from os import environ, listdir, makedirs
+from os.path import join, isdir, exists
+from sys import argv
+from shutil import copy
+
+if "MESON_INSTALL_DESTDIR_PREFIX" in environ:
+    install_dir = environ["MESON_INSTALL_DESTDIR_PREFIX"]
+else:
+    install_dir = environ["MESON_INSTALL_PREFIX"]
+
+include_dir = argv[1] if len(argv) > 1 else "include"
+build_dir = argv[2] if len(argv) > 2 else environ["MESON_BUILD_ROOT"]
+module_dir = join(install_dir, include_dir)
+
+modules = []
+for d in listdir(build_dir):
+    bd = join(build_dir, d)
+    if isdir(bd):
+        for f in listdir(bd):
+            if f.endswith(".mod"):
```

Comment:
While it probably will work with the common compiler, there is no warranty, that a compiler uses the `.mod` extension for the module information files. It would be safer IMO to instruct the compiler to put all module files in a separate directory by the appropriate compiler option and copy the content of that directory, whatever it contains...

Reply:
I think all existing compilers use `.mod` as extension, at least all supported compilers in meson. The only compiler which uses somewhat different rules might be Cray, here the module basename will be in caps rather than lowercase, but I think it still ends with `.mod`.

Unfortunately, meson does not allow to modify the output directory of the build targets, also the output directory name is not stable between version, the depth of the output directory in build tree is known. The main issue with the approach would be a compiler which is not placing the module files in the output directory of the build target (LFortran currently does this).